File: CS Academy/IEEEXtreme 18.0/Stick.cpp
// ﷽
// Contest: IEEEXtreme 18.0
//
// Judge: CS Academy
// URL: https://csacademy.com/ieeextreme18/task/stick/
// Memory Limit: 256
// Time Limit: 500
// Start: Sat 26 Oct 2024 10:21:38 AM EEST
// Reading Time :
// Thinking Time :
// Coding Time :
// Debug Time :
// Submit Count :
// Problem Level :
// Category :
// Comments :
#include <bits/stdc++.h>

#ifdef ALGOAT
#include "debug.hpp"
#else
#define debug(...) 0
#define debug_itr(...) 0
#define debug_bits(...) 0
#endif

// 48-57 -> 0-9  65-90 -> A-Z 97-122 -> a-z
#define fastio()                    \
  ios_base::sync_with_stdio(false); \
  cin.tie(NULL);

#define int long long
#define F first
#define S second
#define all(a) (a).begin(), (a).end()
#define rall(a) (a).rbegin(), (a).rend()
const int dx[8] = {1, 0, -1, 0, 1, 1, -1, -1},
          dy[8] = {0, 1, 0, -1, -1, 1, -1, 1};
using namespace std;

#define MUTLI_CASE
double f(const vector<pair<int,int>>& fig) {
    double res = 0;
    for (unsigned i = 0; i < fig.size(); i++) {
        pair<int,int> p = i ? fig[i - 1] : fig.back();
        pair<int,int> q = fig[i];
        res += (p.first - q.first) * (p.second + q.second);
    }
    return fabs(res) / 2;
}
void solve() {
  int n, k, l;
  cin >> n >> k >> l;
  int y = k + l;
  int x = k - l;
  int d = abs(x - y);
  int area = d * d;

  if (l < k) {
    cout << n*area << endl;
  } else {
  vector<pair<int, int>> arr;
  vector<pair<int, int>> arr1;
  pair<int, int> x1 = {-l, -l};
  pair<int, int> x2 = {-l, l};
  pair<int, int> x3;
  debug(x1);
  debug(x2);
  pair<int, int> prev = x2;

  /*arr.push_back(x1);*/
  for (int i = 1; i < n; i++) {
    x3 = prev;
    x3.F += k;
    arr.push_back(x3);
    debug(x3);
    x3.S += k;
    arr.push_back(x3);
    debug(x3);
    prev = x3;
  }
  for (auto &[a,b] : arr) {
    arr1.push_back({b,a});
  }
  
  arr.push_back({(n-1)*k+l,(n-1)*k+l});
  arr.insert(arr.begin(),{-l,l});
  reverse(all(arr1));
  debug(arr);
  arr1.push_back({l,-l});
  arr1.push_back({-l,-l});
  debug(arr1);
  arr.insert(arr.end(),arr1.begin(),arr1.end());
  debug(arr);
  cout << (int)(f(arr));
  }
  // cout << area << endl;
}
int32_t main() {
  /*freopen("whereami.in", "r", stdin);*/
  /*freopen("whereami.out", "w", stdout);*/
  fastio();
  int n = 1;

#ifdef MUTLI_CASE
  // cin>>n;
#endif
  while (n--) solve();
  return 0;
}

File: CS Academy/IEEEXtreme 18.0/Stick.py
def calculate_area(fig):
    res = 0
    for i in range(len(fig)):
        p = fig[i - 1] if i else fig[-1]
        q = fig[i]
        res += (p[0] - q[0]) * (p[1] + q[1])
    return abs(res) / 2

def solve():
    n, k, l = map(int, input().split())
    
    y = k + l
    x = k - l
    d = abs(x - y)
    area = d * d
    
    if l < k:
        print(n * area)
        return
        
    arr = []
    arr1 = []
    
    x1 = (-l, -l)
    x2 = (-l, l)
    prev = x2
    
    for i in range(1, n):
        x3 = (prev[0] + k, prev[1])
        arr.append(x3)
        x3 = (x3[0], x3[1] + k)
        arr.append(x3)
        prev = x3
    
    for a, b in arr:
        arr1.append((b, a))
    
    arr.append(((n-1)*k + l, (n-1)*k + l))
    arr.insert(0, (-l, l))
    arr1.reverse()
    
    arr1.append((l, -l))
    arr1.append((-l, -l))
    arr.extend(arr1)
    
    print(int(calculate_area(arr)))

# Main execution
if __name__ == "__main__":
    solve()
